File: quarkus-tmdb/deployment/src/test/java/dev/reuss/tmdb/quarkus/deployment/TmdbHealthCheckTest.java
package dev.reuss.tmdb.quarkus.deployment;

import dev.reuss.tmdb.quarkus.runtime.TmdbHealthCheck;
import org.eclipse.microprofile.health.HealthCheck;
import org.eclipse.microprofile.health.Readiness;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

class TmdbHealthCheckTest {

    @Test
    void shouldBeReadinessHealthCheck() {
        assertTrue(HealthCheck.class.isAssignableFrom(TmdbHealthCheck.class));
        assertTrue(TmdbHealthCheck.class.isAnnotationPresent(Readiness.class));
    }
}

File: quarkus-tmdb/deployment/src/test/kotlin/dev/reuss/tmdb/quarkus/deployment/TmdbHealthCheckTest.kt
package dev.reuss.tmdb.quarkus.deployment

import dev.reuss.tmdb.quarkus.runtime.TmdbHealthCheck
import org.eclipse.microprofile.health.HealthCheck
import org.eclipse.microprofile.health.Readiness
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class TmdbHealthCheckTest {
    @Test
    fun shouldBeReadinessHealthCheck() {
        assertTrue(
            HealthCheck::class.java.isAssignableFrom(
                TmdbHealthCheck::class.java,
            ),
        )

        assertTrue(
            TmdbHealthCheck::class.java.isAnnotationPresent(
                Readiness::class.java,
            ),
        )
    }
}
